### PR TITLE
fix: Names in auction end date

### LINKED ISSUE #794: Names in auction end date
### Context information

application version: 
device: 
browser: 
operating system: 

### Steps to reproduce
- Navigate to https://aescan.io/names/meta.chain
- or navigate to https://aescan.io/names/tank.chain

### What is actually happening?

image

The end date ("Ends" row) has a value which is in the past, how can the name still be in auction?

The end date is also not aligned with the one in the "In Auction" table (https://aescan.io/names?type=in-auction):
image

### PR DESCRIPTION
**Pull request by janmichek**
## Description
fixes #794
fixes https://github.com/aeternity/aescan/issues/621

Major refactoring has been done. A detailed A/B check with production would be good.

## Demo
https://github.com/aeternity/aescan/assets/15363559/a7257043-6452-464e-8ee7-47f052888468

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] fix Names in auction end date

### CODE REVIEW

**Comment by Liubov-crypto** [comment]
I'm still seeing meta.chain in auction and expired sections:  https://pr-799-aescan.stg.service.aepps.com/names/meta.chain

could you please check it @janmichek ?
https://github.com/aeternity/aescan/assets/69896204/ed455467-894a-40d8-b9cd-b9d2ec2e71a1

**Comment by janmichek** [comment]
> > I'm still seeing meta.chain in auction and expired sections: https://pr-799-aescan.stg.service.aepps.com/names/meta.chain
> > could you please check it @janmichek ? https://github.com/aeternity/aescan/assets/69896204/ed455467-894a-40d8-b9cd-b9d2ec2e71a1
> 
> Hmm, interesting. The name appears in both lists. I will escalate this to MDW team. For now ignore it.

I created followup on our side to track the issue https://github.com/aeternity/aescan/issues/802

File: src/utils/format.js [hunk 1]
```diff
@@ -101,6 +106,36 @@ export function formatNameStatus(name) {
...
+    name.status === 'auction'
+
+  const isExpired = () =>
+    name.status === 'name' && !name.active
+
+  const isRevoked = () =>
+    isExpired() && name.active === false && name.info.expireHeight + REVOKED_PERIOD > blockHeight
+
+  const isActive = () =>
+    name.active
```
Comment (michele-franchi) [inline src/utils/format.js:120]: Shouldn't these be simple variables instead of functions?

File: src/components/NameDetailsPanel.vue [hunk 2]
```diff
@@ -99,41 +97,41 @@
...
             <hint-tooltip>
-              {{ namesHints[nameStatusHint + 'Height'] }}
+              {{ namesHints[state + 'Height'] }}
             </hint-tooltip>
           </th>
           <td class="name-details-panel__data">
             <app-link
-              :to="`/keyblocks/${name.expirationHeight}`">
-              {{ name.expirationHeight }}
+              :to="`/keyblocks/${ name.acutionEndsHeight || name.expirationHeight}`">
```
Comment (michele-franchi) [inline src/components/NameDetailsPanel.vue:107]:
```suggestion
              :to="`/keyblocks/${name.acutionEndsHeight || name.expirationHeight}`">
```
Just an extra space

**Review by michele-franchi — CHANGES_REQUESTED** [review]
Looks good but the code can be slightly improved.

File: src/utils/format.js [hunk 3]
```diff
@@ -101,6 +106,30 @@ export function formatNameStatus(name) {
   }
 }
 
+export function formatNameState(name, blockHeight) {
+  const isInAuction = name.status === 'auction'
+  const isActive = () => name.active
```
Comment (michele-franchi) [inline src/utils/format.js:111]:
```suggestion
  const isActive = name.active
```

**Review by michele-franchi — APPROVED** [review]
Just an extra function but other than that looks good.

**Pull request merged**